### PR TITLE
Fix Create Environment agent version for Serverless

### PR DESCRIPTION
**Pull request by opauloh**
### Summary of your changes

This fixes https://github.com/elastic/cloudbeat/issues/2546, by using the version inputted by the user in the Create Environment parameters instead of using the version from the `env.ELK_VERSION`.

While it is true that the version doesn't matter on Serverless in terms of Kibana and Elasticsearch, it does matter for deploying agents as the snapshots might be no longer available.

### CODE REVIEW

**Review by gurevichdmitry — APPROVED** [review]

**Comment by opauloh** [comment]
@gurevichdmitry it seems I can't merge this PR even after it was approved, could you merge it?

**Pull request merged**
